lexer.py

```diff
@@ -0,0 +1,142 @@
+from tokens import TokenTypes, Token
+
+###############################################################################
+#                                                                             #
+#   LEXER                                                                     #
+#                                                                             #
+###############################################################################
+
+ESCCHARS = {
+    "n":  "\n",
+    "t":  "\t",
+    "\\": "\\",
+    "'": "'",
+    "r":  "\r",
+}
+
+RESERVED_KEYWORDS = {
```

Comment:
Maybe do:
```python
RESERVED_KEYWORDS = ['SELECT', 'FROM', ...]
KEYWORD_CONSTRUCTORS = {kword: (lambda x: Token(TokenTypes[kword], kword, x)) for kword in RESERVED_KEYWORDS}
```

I notice that this won't work for `TRUE` and `FALSE` -- but, actually, why do we need to store `token_value` in the first place?  The only place where it seems like it's actually needed is the `ID` case.

Reply:
On further thought, I feel like these constructors belong in the `Token` class -- they feel a little out of place here.

Reply:
We also need `token_value` for numbers and strings but I see your point.